autorest/models/schema.py

```diff
@@ -24,14 +24,93 @@
+
+class ComposedProperty:
+    """Represents a property of a CompositeType
     """
+    def __init__(self, name, description, property_type, **kwargs):
+        self.name = name
+        self.description = description
+        self.property_type = property_type
+        self.required = kwargs.pop('required', False)
+        self.readonly = kwargs.pop('readonly', False)
+        self.constant = kwargs.pop('constant', False)
+        self.default_value = kwargs.pop('default_value', None)
+
+    def get_property_documentation_string(self):
+        if not self:
+            raise TypeError("Property can not be none.")
+        if self.constant or self.readonly:
+            doc_string = ":ivar {}:".format(self.name)
+        else:
+            doc_string = ":param {}:".format(self.name)
+        if self.required:
+            doc_string += " Required."
+
+        description = self.description
+        if description and description[-1] != ".":
+            description += "."
+
+        # documentation = self.documentation
+        # if documentation:
+        #     if documentation[-1] != ".":
+        #         documentation += "."
+        #     documentation += " Default value: " + self.default_value + " ."
+        if description:
+            doc_string += " " + description
+        if self.default_value:
+            doc_string += " Default value: \"" + self.default_value + "\"."
+        # if documentation:
+        #     doc_string += " " + documentation
+        return doc_string
+
 
-    def __init__(self, name):
+class CompositeType:
+    """Represents a composite type schema ready to be serialized in Python.
+    """
+    def __init__(self, name, description, **kwargs):
         self.name = name
+        self.description = description
+        self.properties = kwargs.pop('properties', None)
+
+    @classmethod
+    def _create_properties(cls, yaml_data):
+        # Returns the properties of a CompositeType.
+        properties = []
+        for k, v in yaml_data['properties'].items():
+            required = v['details']['default']['required']
+            readonly = v['details']['default']['readOnly']
+            constant = v['details']['default'].get('constant', False)
+            property_type = v['schema']['type']
+            if property_type == 'object':
+                property_type = v['schema']['details']['default']['name']
+
+            # what if list, dict
+
+            composed_property = ComposedProperty(
+                name=k,
+                description=v['details']['default']['description'].strip(),
+                property_type=property_type,
+                required=required,
+                readonly=readonly,
+                constant=constant,
+                default_value=v['schema'].get('default')
+            )
+            properties.append(composed_property)
+        properties.sort(key=lambda item: (not(item.readonly or item.constant), item.name))
```

Comment:
Does it support your inheritance ordering already?

Reply:
yeah, it appears that it does so I'll get rid of this section. thanks!